### PR TITLE
Update devenv-data-transfer.md

### PR DESCRIPTION
**Pull request by NKarolak**
Added AL example for replacing a ModifyAll.

### CODE REVIEW

File: dev-itpro/developer/devenv-data-transfer.md [hunk 1]
```diff
@@ -138,6 +138,24 @@ begin
 end;
 ```
 
+### Example 2 - replace ModifyAll()
```
Comment (PhDuck) [inline dev-itpro/developer/devenv-data-transfer.md:141]: Could you maybe move this part just after the section "Uniqueness in the source table" but before "CopyRows". It seems the "Performance" and "Uniqueness in the source table" sections pertains to the above example.

File: dev-itpro/developer/devenv-data-transfer.md [hunk 2]
```diff
@@ -138,6 +138,24 @@ begin
...
+    dest: Record Destination;
+begin
+    dt.SetTables(Database::Destination, Database::Destination);
+    dt.AddSourceFilter(dest.FieldNo("Field 1"), '=%1', 'A');
+    dt.AddSourceFilter(dest.FieldNo("Field 2"), '%1..%2', 'B', 'C');
+    dt.AddConstValue(dest."Enum Field"::SomeValue, dest.FieldNo("Enum Field"));
+    dt.CopyFields();
+end;
+```
+
```
Comment (PhDuck) [inline dev-itpro/developer/devenv-data-transfer.md:158]: Could you include some motivation for wanting to replace ModifyAll with this? What is the positives (and negatives).

**Review by PhDuck — COMMENTED** [review]
Please restructure and explain why one should use this approach rather than ModifyAll.

**Comment by NKarolak** [comment]
> Please restructure and explain why one should use this approach rather than ModifyAll.

I thought I could omit that because under Design Guidelines, you already specified:

> Because DataTransfer operates in bulk and not on a row-by-row basis, row based events or triggers won't be executed. For example, when calling CopyFields, none of the following events will be called: OnBeforeModify, OnModify, or OnAfterModify. Or, when calling CopyRows, none of the following events will be called: OnBeforeInsert, OnInsert, or OnAfterInsert.

Do I still need to change that?

File: dev-itpro/developer/devenv-data-transfer.md (hunk 1 above)
Comment (NKarolak) [reply dev-itpro/developer/devenv-data-transfer.md:158]: Good point, updated.

File: dev-itpro/developer/devenv-data-transfer.md (hunk 2 above)
Comment (NKarolak) [reply dev-itpro/developer/devenv-data-transfer.md:175]: Same answer as for your first comment, please check.

**Review by PhDuck — APPROVED** [review]

**Comment by SusanneWindfeldPedersen** [comment]
Thanks for your contribution, much appreciated!

**Pull request merged**